algebra/csc_tools/csc_utils.c

```diff
@@ -1,5 +1,32 @@
-#include "cs.h"
 
+#ifndef EMBEDDED
+
+#include "csc_utils.h"
+
+//========== Logical, testing and debug ===========
+
+c_int csc_is_eq(csc *A, csc *B, c_float tol) {
+  c_int j, i;
+
+  // If number of columns does not coincide, they are not equal.
+  if (A->n != B->n) return 0;
+
+  for (j = 0; j < A->n; j++) { // Cycle over columns j
+    // if column pointer does not coincide, they are not equal
+    if (A->p[j] != B->p[j]) return 0;
+
+    for (i = A->p[j]; i < A->p[j + 1]; i++) { // Cycle rows i in column j
+      if ((A->i[i] != B->i[i]) ||             // Different row indices
+          (c_absval(A->x[i] - B->x[i]) > tol)) {
```

Comment:
I think you are right.   Perhaps the easiest fix is to compare the index of the last entry of each column, rather than the first.   As written I think the bad case is when the matrices are identical except in the last element of the `p` field.

I will make a change but keep it open, let me know what you think.